### DISCUSSION TITLE
New features [poll]

### DISCUSSION THREAD

**Discussion by Valdes-Tresanco-MS** [Valdes-Tresanco-MS/gmx_MMPBSA#49] [Ideas]
We have a few new features that we want to implement. However, we also want to consider the needs of our users.

We will describe each of the characteristics in different threads. You can rate the ones of your preference or add a new one.

_For the future we will take into account the highest scoring_
**Help us improve gmx_MMPBSA!**

**Comment by Valdes-Tresanco-MS** [comment]
~Multi-residue Alanine scanning [in progress]~

**Reply by Valdes-Tresanco-MS** [reply]
Discarded.

**Comment by Valdes-Tresanco-MS** [comment]
Consideration of explicit water molecules around the receptor-ligand interface

**Comment by Valdes-Tresanco-MS** [comment]
MMPBSA.py support

**Comment by Valdes-Tresanco-MS** [comment]
NAMD support

**Comment by Valdes-Tresanco-MS** [comment]
Calculate energy and errors with different methods (i.e. bootstrap) [in progress]

**Comment by Valdes-Tresanco-MS** [comment]
Resume calculations after an interruption

**Comment by Valdes-Tresanco-MS** [comment]
Free energy surface for stability calculation

**Comment by Gsk468** [comment]
Probability density function (PDF), PCA and DSSP in gmx_mmpbsa_ana module.

**Comment by yoavatsmonraz** [comment]
ddg_mutation to other residues besides alanine and glycine

**Reply by Valdes-Tresanco-MS** [reply]
That would be out of our hands, at least for the moment. At some point, we started working on a backrub for this purpose, however, this is very complicated to do. Any mutant that is larger than the size you are mutating would be a considerable topological change. This would make it necessary to re-couple the receptor and ligand again since an energy minimization would not be clear whether it would converge or not. In addition, there is also no certainty that each frame undergoing this process would be consistent with the previous one, making it possible to obtain a very different conformation from one frame to the next. This would require not only programming but also testing and validation of its operation. This is an ambitious project that is out of our hands. In any case, if there is someone interested in carrying it out, we will gladly help you, we have already looked into this matter and we have some ideas.

**Comment by Jameel9** [comment]
Hello Sir

I hope you are doing great

I think it would be a good idea If you can use your Github contents to write a clear and detailed tutorial for gmx_MMPBSA to be published in the _Living Journal of Computational Molecular Simulation_ as Dr. Justin did for his gromacs tutorials. 
It will help new learers a lot.

**Reply by Valdes-Tresanco-MS** [reply]
Thank you @Jameel9 for your suggestion, we will take it into account. We are working on improving first the gmx_MMPBSA_ana since its current performance is poor. Then we will implement two new methods and extend the calculations to several programs. That is why we have decided to postpone the improvement of the tutorials, in order to do it at once so that there are no abrupt changes between the tutorials published in the documentation and a future article.

**Comment by ShivBharadwaj** [comment]
Hello!

Packages for calculating BE for DNA-DNA and  DNA-RNA, RNA-Protein, and DNA/RNA-Small molecule complexes can be considered for improvements.  

Thank you!

**Reply by marioernestovaldes** [reply]
In theory, gmx_MMPBSA can be used to calculate the binding free energy of those systems

**Reply by ShivBharadwaj** [reply]
For example, I see the package supports Protein-DNA traj. Do you think, I can use the same example for calculating BE for DNA-Small molecules and other with some modifications in the system?

**Reply by marioernestovaldes** [reply]
Sure! you can use gmx_MMPBSA for that and any other combination of molecules

**Reply by ShivBharadwaj** [reply]
Okie, I will try it on DNA-Small molecule gro traj. Thanks!

**Comment by laodeaman-ai** [comment]
Hello Sir...

I hope you are doing great

I am encountering an error when executing the following command using gmx_MMPBSA version 1.6.1 on Ubuntu 20.04 with gmx 2022.5 and ambertools21.12:

gmx_MMPBSA -O -i gb.in -cs ../md.tpr -ct ../md.xtc -ci ../index.ndx -cg 1 13 -cp ../topol.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv

After running the above command, I receive the following error message:

[INFO ] Checking gb.in input file...
[INFO ] Checking gb.in input file...Done.

[INFO ] Checking external programs...
[INFO ] cpptraj found! Using /home/arga/miniconda3/envs/gmxMMPBSA/bin/cpptraj
[INFO ] tleap found! Using /home/arga/miniconda3/envs/gmxMMPBSA/bin/tleap
[INFO ] parmchk2 found! Using /home/arga/miniconda3/envs/gmxMMPBSA/bin/parmchk2
[INFO ] sander found! Using /home/arga/miniconda3/envs/gmxMMPBSA/bin/sander
[INFO ] Using GROMACS version 4.x.x!
[ERROR ] MMPBSA_Error

I would greatly appreciate any guidance or solution to resolve this error. Thank you for your assistance.

**Reply by Valdes-Tresanco-MS** [reply]
Please, report it in a new [issue](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/issues/new/choose)

**Reply by laodeaman-ai** [reply]
Ok. Thanks
